js/packages/mobile-wallet-adapter-walletlib/android/src/main/java/com/solanamobile/mobilewalletadapterwalletlib/reactnative/SolanaMobileWalletAdapterWalletLibModule.kt

```diff
@@ -159,147 +142,195 @@ class SolanaMobileWalletAdapterWalletLibModule(val reactContext: ReactApplicatio
 
     /* Generic Request functions */
     @ReactMethod
-    fun cancelRequest() {
-        Log.d(TAG, "Cancelled request");
-        (request as? ScenarioRequest)?.let { scenarioRequest ->
+    fun cancelRequest(sessionId: String, requestId: String) {
+        Log.d(TAG, "Cancelled request $requestId");
+        (pendingRequests.remove(requestId) as? ScenarioRequest)?.let { scenarioRequest ->
             scenarioRequest.cancel();
         }
     }
 
     /* AuthorizeDapp Request */
     @ReactMethod
-    fun completeWithAuthorize(publicKey: ReadableArray, accountLabel: String?, walletUriBase: String?, authorizationScope: ReadableArray?) {
-        Log.d(TAG, "completeWithAuthorize: authorized public key = $publicKey")
-        (request as? MobileWalletAdapterRemoteRequest.AuthorizeDapp)?.request?.let { authRequest ->
-            authRequest.completeWithAuthorize(
-                convertFromReactByteArray(publicKey),
-                accountLabel,
-                null, // walletUriBase,
-                null, // convertFromReactByteArray(authorizationScope)
-            )
+    fun completeWithAuthorize(
+        sessionId: String, 
+        requestId: String, 
+        publicKey: ReadableArray, 
+        accountLabel: String?, 
+        walletUriBase: String?, 
+        authorizationScope: ReadableArray?
+    ) = launch {
+        checkSessionId(sessionId) {
+            Log.d(TAG, "completeWithAuthorize: authorized public key = $publicKey")
+            (pendingRequests.remove(requestId) as? MobileWalletAdapterRemoteRequest.AuthorizeDapp)?.request?.let { authRequest ->
+                authRequest.completeWithAuthorize(
+                    publicKey.toByteArray(),
+                    accountLabel,
+                    null, // walletUriBase,
```

Comment:
TODO: Fill in these `null` fields.

`walletUriBase` is the custom "scheme" wallets can return to dapps to use (not sure if anyone actually uses this lol)
`authorizationScope` is the auth key that we implement during ClientTrustUseCase. Maybe we can just add a dummy String/bytearray?

Reply:
agreed, tho lets tackle in a separate PR